core/src/main/scala/org/apache/spark/ui/jobs/JobPage.scala

```diff
@@ -32,7 +32,12 @@ private[ui] class JobPage(parent: JobsTab) extends WebUIPage("job") {
 
   def render(request: HttpServletRequest): Seq[Node] = {
     listener.synchronized {
-      val jobId = request.getParameter("id").toInt
+      val paramaterId = request.getParameter("id")
+      if (paramaterId.equals("")) {
```

Comment:
Yes, I updated the code to handle both cases.

BTW I test it only by hand because I not found any testcase for HTTP UI.